### ISSUE TITLE
Simplify JS module structure

### ISSUE THREAD

**Issue by willeastcott** [playcanvas/engine#2151]
Current JS module structure in PlayCanvas is needlessly verbose:

```
Object.assign(pc, function () {
    'use strict';

    var SomeClass = function () {
    };

    Object.assign(SomeClass.prototype, {
        someFunc: function () {
        }
    });

    return {
        SomeClass: SomeClass
    };
}());
```

There are currently 281 JS files in the engine source tree. Almost all follow this pattern. That's a lot of needless calls to Object.assign and closure creations.

I suggest transforming this to:

```
function SomeClass() {
}

Object.assign(SomeClass.prototype, {
    someFunc: function () {
    }
});
```

Additionally, we should remove all internal code use of the `pc.` namespace, since inside the engine, this is unnecessary. So instead of:

```
var v = new pc.Vec3();
```

Do:

```
var v = new Vec3();
```

Once this is done, we could consider using the ES6 module `import`/`export` directives to work alongside a bundler to facilitate tree shaking. Plus building the whole engine as a JS module.

**Comment by slimbuck** [comment]
We are currently able to expose interfaces and functions selectively to the pc namespace (via the return clause). This means we can have local variables declared that don't pollute the pc namespace. How do you see us handling these internal variables in the new system?

**Comment by willeastcott** [comment]
We'd use `export` only on API objects/properties. Anything else would be inside the library's closure and private to the engine source.

**Comment by Maksims** [comment]
If script is loaded as a module using ES6 `import`, then all internal code is local to that module, and only what is exported will be available.
But, it only works well if they are separate files. Of course when engine is built to single file, this doesn't work. I couldn't find ES6 module info on concatenation.

Alternative to this:
```javascript
Object.assign(SomeClass.prototype, {
    someFunc: function () {
    }
});
```

Is this:
```javascript
SomeClass.prototype.someFunc = function () {
};
```

But I think it is matter of taste and really not changing a situation here.

Going from namespace to non-namespace globally - is a risky move. That can lead to names clashing, especially for existing projects.

**Comment by willeastcott** [comment]
Fixed by #2166
